cmd/agent/api/grpc.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+/*
+Package api implements the agent IPC api. Using HTTP
+calls, it's possible to communicate with the agent,
+sending commands and receiving infos.
+*/
+package api
+
+import (
+	"context"
+
+	pb "github.com/DataDog/datadog-agent/cmd/agent/api/pb"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	hostutil "github.com/DataDog/datadog-agent/pkg/util"
+)
+
+type server struct {
+	pb.UnimplementedAgentServer
```

Comment:
A bit too late for this but I'm not entirely sure to understand this naming, is it about compatibility with Agent not implementing the gRPC server?

Reply:
The UnimplementedXXXXXXServer is always generated by protobuf and implements the service interface, developers implementing the service interface can embed it to ensure new methods can be added in forward compatible manner. You can take a look at the generated code here: https://github.com/DataDog/datadog-agent/blob/b13aba568f2de49b307ec8ba4c0cefc3d4d60d34/cmd/agent/api/pb/api.pb.go#L248-L254